src/Symfony/Component/Serializer/Normalizer/JsonSerializableNormalizer.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Normalizer;
+
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+
+/**
+ * A normalizer that uses an objects own JsonSerializable implementation.
+ *
+ * @author Fred Cox <mcfedr@gmail.com>
+ */
+class JsonSerializableNormalizer implements NormalizerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function normalize($object, $format = null, array $context = [])
+    {
+        if (!$object instanceof \JsonSerializable) {
+            throw new InvalidArgumentException('The object must implement "\JsonSerializable".');
```

Comment:
Like this?

`throw new InvalidArgumentException("The object must implement " . \JsonSerializable::class);`